lib/AST/IRToASTVisitor.cpp

```diff
@@ -505,20 +459,57 @@ void IRToASTVisitor::VisitFunctionDecl(llvm::Function &func) {
 
   tudecl->addDecl(decl);
 
-  if (func.arg_empty()) {
-    return;
-  }
-
   std::vector<clang::ParmVarDecl *> params;
   for (auto &arg : func.args()) {
     auto parm{clang::cast<clang::ParmVarDecl>(GetOrCreateDecl(&arg))};
     params.push_back(parm);
   }
 
-  decl->getAsFunction()->setParams(params);
+  auto fdecl{decl->getAsFunction()};
+  fdecl->setParams(params);
+
+  for (auto &inst : llvm::instructions(func)) {
```

Comment:
Can we do this in `BlockVisitor` as we create instructions? I'd prefer `VisitFunctionDecl` to not create AST nodes that relate to the function body.

Reply:
Things are easier if we can create all declarations at once, and to me `VisitFunctionDecl` looks like the best place to do so right now...